Grammar generator

ldez opened the pull request:
- [x] merge PR #58 and #60 
- [x] split the grammars file into multiple files (partials grammars)
- [x] create a master grammar
- [x] create a grammars generator (use master grammar and partials grammars)
- [x] create a grammars generator for the code blocks (Asciidoctor and Markdown) 
- [x] Document generation system. (and GitHub PR template)

I've created a package for gitignore's like file with a small generator (https://github.com/ldez/atom-language-ignore).
I'm going to use this as starting point.

ldez commented:
@nicorikken you can try the new system. :smile:

ldez commented:
@nicorikken and @mojavelinux It is ready!

I add the change from the #67 (fix #65)

nicorikken commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,30 @@
+
+## Syntax example
+
+Your syntax exemple.
```
Typo?

nicorikken commented on grammars/repositories/languages.cson:
```diff
@@ -0,0 +1,79 @@
+[
```
Maybe order this alphabetically?

ldez replied:
Done!

nicorikken commented:
Whow, this is amazing stuff!!! :+1: :beers: :clap: Have to reorient myself somewhat though. 
Some remarks:
- [ ] Order the languages alphabetically in `grammars/repositories/languages`
- [ ] Compile with the latest grammar, to include the `table` and `various-blocks`
- [ ] Document the `season` dependency somewhere (at least, that was the one that I was missing) Maybe just `npm install` ?

As this is quite a breaking change, I opt for getting this merge next, and implement/rewrite all other PR's on top of this PR.

ldez commented:
I don't understand:
> Compile with the latest grammar, to include the table and various-blocks

For installing the dependencies do: `apm install`

nicorikken commented:
> For installing the dependencies do: `apm install`

I didn't know that actually... :disappointed: :+1:

ldez commented:
- [x] Order the languages alphabetically in grammars/repositories/languages
- [x] Document the `season` dependency somewhere

nicorikken commented:
Somehow when I recompiled the grammar last time, it did result in the two sections added on top of the checkout. Now it seems to be fine. I'm up for a merge!

ldez commented:
You can also close  #36 and #67  :wink:

ldez commented:
@nicorikken  can you make a release?

use:
- `apm publish major`: 0.12.0 -> 1.0.0
- or `apm publish minor`: 0.12.0 -> 0.13.0

Those commands do automatically a tag on Git.

nicorikken commented:
Merged! Some improvements here and there will probably still be required, but now we have a new structure to base the reset of the development on.

nicorikken commented:
@ldez I was aiming to get my bold fix incorporated, and maybe rewrite my previous italic PR (which is too greedy). What do you think?

ldez commented:
I think it's releasing time !

I am looking forward to use all the new feature outside developer mode.
There are already a lot of changes since the last release.

But if you want to add the bold fix, I can wait :wink:

nicorikken commented:
Alright, working on it! Same here, can't wait to share these improvements with my colleagues in a regular package update.

mojavelinux commented:
I think we should move properties-lang.cson into a subdirectory, perhaps "contrib" or "alt". I think the visibility of the file is too high when adjacent to asciidoctor-grammar.cson.

nicorikken commented:
Good catch. I'v opened it up accidentally as well.

ldez commented:
we cannot do that, the file grammar must be in a first level of `grammars` folder.

mojavelinux commented:
I see. Can it be named `language-properties.cson` instead of `properties-lang.cson`?

ldez commented:
yes :+1:

ldez commented:
Fix with  #101

mojavelinux commented:
\o/
